### PR TITLE
Closes #771: Use system branding block to support text-based web headers and more.

### LINKED ISSUE #771: Create text logo option
Match QS1 settings

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
This is an alternate PR for #771 adapted from @jwmoore1's #1054.

This approach fully embraces the system branding block added in D8 core and does not add any additional theme settings to support text-based web headers.

Full list of changes:
- Fleshed out `block--system-branding-block.html.twig` with full support for logo, site name and site slogan
- Added a `branding` theme region
- Enabled system branding block in branding region (with only the logo enabled) by default
- Added site slogan to Quickstart settings page
- Replaced links to site information page in system branding block with links to Quickstart settings page
- Reverted some of @jwmoore1's changes to the theme settings, etc
- Added DB update to configure and enable branding block as well as new theme region settings on existing sites

Screen Shot 2021-09-27 at 2 01 20 PM
Screen Shot 2021-09-27 at 4 07 43 PM
Screen Shot 2021-09-27 at 4 08 37 PM

## Related Issue
#771

## How Has This Been Tested?

Locally with lando.  DB update still needs testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_core/az_core.module [hunk 1]
```diff
@@ -162,3 +162,38 @@ function az_core_canonical_url(array $data) {
 
   return $uri;
 }
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter().
```
Comment (joeparsons) [inline modules/custom/az_core/az_core.module:167]:
```suggestion
 * Implements hook_form_FORM_ID_alter().
```

**Review by danahertzberg — CHANGES_REQUESTED** [review]

**Comment by mmunro-ltrr** [comment]
Trying a comparison of this and #1054. Having the branding block to display text headers gives easy access (just clicking on the usual pencil icon is enough). A potential source of confusion is the block title, since that's up at the top of the editing form, with a checkmark to enable the display (which works!), in addition to the options to display or enter the actual site title. The block display options are fully functional, & very powerful (you can do things like suppressing the site name display on selected content types). Looks good, but might need more documentation to prevent people doing weird things.

**Review by camikazegreen — APPROVED** [review]

**Review by mmunro-ltrr — DISMISSED** [review]

**Review by tadean — DISMISSED** [review]

File: az_quickstart.profile [hunk 2]
```diff
@@ -28,3 +28,23 @@ function az_quickstart_update_9203() {
...
+    ->set('bootstrap_barrio_region_class_branding', '')
+    ->save(TRUE);
+
+  $config = \Drupal::service('config.factory')->getEditable('block.block.az_barrio_branding');
+  $config
+    ->set('status', TRUE)
+    ->set('region', 'branding')
+    ->set('weight', -9)
+    ->set('use_site_name', FALSE)
+    ->set('use_site_slogan', FALSE)
```
Comment (trackleft) [inline az_quickstart.profile:48]:
```suggestion
    ->set('settings.use_site_name', FALSE)
    ->set('settings.use_site_slogan', FALSE)
```

File: modules/custom/az_core/az_core.module [hunk 3]
```diff
@@ -162,3 +162,38 @@ function az_core_canonical_url(array $data) {
...
+      $quickstart_settings_url = $url_quickstart_settings->toString();
+
+      // Provide link to Quickstart settings page if the user has access.
+      $site_name_description = t('Defined on the <a href=":information">Quickstart settings</a> page.', [':information' => $quickstart_settings_url]);
+      $site_slogan_description = t('Defined on the <a href=":information">Quickstart settings</a> page.', [':information' => $quickstart_settings_url]);
+    }
+    else {
+      // Explain that the user does not have access to the Quickstart settings
+      // page.
+      $site_name_description = t('Defined on the Quickstart settings page. You do not have the appropriate permissions to change the site logo.');
```
Comment (joeparsons) [inline modules/custom/az_core/az_core.module:192]:
```suggestion
      $site_name_description = t('Defined on the Quickstart settings page. You do not have the appropriate permissions to change the site name.');
```

File: modules/custom/az_core/az_core.module [hunk 4]
```diff
@@ -162,3 +162,38 @@ function az_core_canonical_url(array $data) {
...
+      $site_slogan_description = t('Defined on the Quickstart settings page. You do not have the appropriate permissions to change the site logo.');
```
Comment (joeparsons) [inline modules/custom/az_core/az_core.module:193]:
```suggestion
      $site_slogan_description = t('Defined on the Quickstart settings page. You do not have the appropriate permissions to change the site slogan.');
```

**Review by trackleft — APPROVED** [review]

**Comment by joeparsons** [comment]
The database update seems to be working as expected now.

**Review by trackleft — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Review by mmunro-ltrr — DISMISSED** [review]

**Review by smayden — APPROVED** [review]

File: themes/custom/az_barrio/templates/blocks/block--system-branding-block.html.twig [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * - site_slogan: Slogan for site as defined in Site information settings.
+ */
+#}
+{% block content %}
+  {% if site_logo %}
+    {{ site_logo }}
+  {% endif %}
+  {% if site_name %}
+    <a class="d-flex align-items-center justify-content-start text-decoration-none" href="{{ path('<front>') }}" title="{{ 'Home'|t }}" rel="home">
+      <span class="d-inline-block h3 text-blue text-uppercase bold my-3">{{ site_name }}</span>
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/blocks/block--system-branding-block.html.twig:22]:
```suggestion
      <span class="d-inline-block h3 text-blue text-uppercase bold my-4">{{ site_name }}</span>
```

File: themes/custom/az_barrio/templates/blocks/block--system-branding-block.html.twig [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    </a>
+  {% endif %}
+  {% if site_slogan %}
+    <div class="d-inline-block align-top site-name-slogan lead mb-3">
```
Comment (danahertzberg) [inline themes/custom/az_barrio/templates/blocks/block--system-branding-block.html.twig:26]:
```suggestion
    <div class="d-inline-block align-top site-name-slogan lead mb-4">
```

**Review by danahertzberg — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Review by mmunro-ltrr — APPROVED** [review]

**Pull request merged**